application/src/test/java/org/thingsboard/server/controller/BaseDashboardControllerTest.java

```diff
@@ -495,4 +514,107 @@ public void testAssignDashboardToEdge() throws Exception {
         Assert.assertEquals(0, pageData.getData().size());
     }
 
+    @Test
+    public void testDeleteDashboardExistsRelation_Error_RestoreRelation_DeleteRelation_DeleteDashboard_Ok() throws Exception {
+        Dashboard dashboard = new Dashboard();
+        dashboard.setTitle("My dashboard");
+        Dashboard savedDashboard = doPost("/api/dashboard", dashboard, Dashboard.class);
+
+        Alarm savedAlarm = createAlarm(savedDashboard.getId());
```

Comment:
Copy-paste the method `createAlarm()` from another class is not the best practice. 
Probably it is enough to call just alarmService.createOrUpdateAlarm(savedAlarm)

Reply:
OK